packages/store-sync/src/recs/syncToRecs.ts

```diff
@@ -59,9 +64,46 @@ export async function syncToRecs<TConfig extends StoreConfig = StoreConfig>({
 
   world.registerDisposer(stopSync);
 
+  const getResourceSelector = async (functionSelector: Hex): Promise<Hex> => {
+    const entity = encodeEntity<
+      KeySchema & {
+        functionSelector: "bytes4";
+      }
```

Comment:
why do we need this? I would expect it to be inferred from `components.FunctionSelectors.metadata.keySchema`

Reply:
`SchemaToPrimitives` didn't seem to pick up the inferred type but it works now. Will remove